syntax = "proto2";
package tdrec.protos;

message DatasetConfig {
    required DatasetType dataset_type = 1 [default = CsvDataset];

    // delimiter of column features, only used for CsvDataset
    optional string delimiter = 2 [default = ","];

    // mini batch size to use for training and evaluation.
    optional uint32 batch_size = 3 [default = 1024];

    // the input fields must be the same number and in the
    // same order as data in csv files
    repeated Field input_fields = 4;

    // labels
    repeated string label_fields = 5;

    // sample weights
    optional string sample_weight_field = 6;
}

enum DatasetType {
    CsvDataset = 1;
    ParquetDataset = 2;
}

message Field {
    required string input_name = 1;
    required FieldType input_type = 2;
    optional uint32 split_length = 3;
    optional string sub_type = 4;
}

enum FieldType {
    DOUBLE = 1;
    INT = 2;
    STRING = 3;
}
